### DISCUSSION TITLE
Execute unit test

### DISCUSSION THREAD

**Discussion by samareshdasglobal** [TerriaJS/terriajs#7061] [Q&A] [answered]
Hello,
I've introduced a new component to TerriaJS and have implemented unit tests for it, aligning with the existing approach. However, I haven't identified any script for running tests in the package.json file. To execute all tests, what modifications should I make to the configuration?

**Comment by zoran995** [comment]
As described in docs https://docs.terria.io/guide/contributing/development-environment/#tests-specs 
from inside terrirajs repo build terriajs and then
```sh
yarn start
```

**Reply by samareshdasglobal** [reply]
Thanks for your response. According to documentation it needs to build the terriajs first. When try to build the terrijs by yarn gulp i am getting following error
✖ 568 problems (63 errors, 505 warnings)
  3 errors and 0 warnings potentially fixable with the `--fix` option.

[17:12:40] 'lint' errored after 13 s
[17:12:40] Error in plugin "eslint/bin/eslint.js"
Message:
    External module exited with an error.
Details:
    domainThrown: true

[17:12:40] 'default' errored after 13 s
error Command failed with exit code 1.
/home/samaresh/Development/Map Applications/gt-gis-mapping/packages/terriajs/lib/ReactViews/Search/SearchBox.jsx
  153:9   error  Unknown property 'css' found  react/no-unknown-property
  160:11  error  Unknown property 'css' found  react/no-unknown-property

when try to execute yarn start and browse http://localhost:3002/SpecRunner.html, it is showing no spec found

**Reply by nf-s** [reply, marked as answer]
Hi @samareshdasglobal 

It looks like you are hitting some `lint` errors - can you please try running `yarn gulp build` or `yarn gulp watch` instead?
